### PR TITLE
ext: simplelink: Restructure CC3220SDK as SimpleLink SDK

### PR DESCRIPTION
**Pull request by GAnthony**
The SimpleLink SDK actually encapsulates a family of devices,
of which cc32xx is just one device.  Other devices can fit
under this SDK directory structure.

This expansion will also allow the import of the WiFi
host driver and its driver porting layer in the future,
inserted at the correct levels in the SimpleLink SDK
directory heirarchy.

Follow the URL (ending in "#directory-structure") referenced in
ext/hal/ti/simplelink/README which explains the SDK structure,
and where devices fit in.

Jira: ZEP-1958

Change-Id: I16515d3e3779de0d55d3b1b8e25029609d1f66c3
Signed-off-by: Gil Pitney <gil.pitney@linaro.org>

---
This change is [Reviewable](https://reviewable.io/reviews/zephyrproject-rtos/zephyr/35)

### CODE REVIEW

File: ext/hal/ti/simplelink/Makefile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+ifdef CONFIG_HAS_CC3220SDK
+ZEPHYRINCLUDE +=-I$(srctree)/ext/hal/ti/simplelink/source/ti/devices/cc32xx
+ZEPHYRINCLUDE +=-I$(srctree)/ext/hal/ti/simplelink/source/ti/devices/cc32xx/inc
+ZEPHYRINCLUDE +=-I$(srctree)/ext/hal/ti/simplelink/source/ti/devices/cc32xx/driverlib
+KBUILD_CFLAGS += -DUSE_CC3200_ROM_DRV_API
```
Comment (xusiwei) [inline ext/hal/ti/simplelink/Makefile]: @GAnthony do you means that, we are deprecate the old cc3200sdk support ?

Comment (GAnthony) [inline ext/hal/ti/simplelink/Makefile]: Yes, the cc3200sdk will be deprecated with the cc3200 board.
However, I just noticed something about KBUILD_CFLAGS.  This patch still works with the CC3220, but to be more correct, it should be KBUILD_CFLAGS += -DUSE_CC3220_ROM_DRV_API.

**Review by galak — CHANGES_REQUESTED** [review]
Lets make the fixup the Kbuild.

**Comment by GAnthony** [comment]
Fixed the Makefile to use KBUILD_CFLAGS=-DUSE_CC3220_ROM_DRV_API, tested on Zephyr samples, and re-pushed the commit.

File: ext/hal/ti/simplelink/Makefile (hunk 1 above)
Comment (xusiwei) [inline ext/hal/ti/simplelink/Makefile]: @GAnthony  thank you

**Comment by GAnthony** [comment]
Not sure why getting "Shippable" failures on Mr. Chettimada's commit messages, when my commit has no such error.   
Should I close and re-submit (to master branch)?

https://app.shippable.com/github/zephyrproject-rtos/zephyr/runs/200/1/tests

**Comment by galak** [comment]
No please leave this, I'm trying to figure out why this happened myself.  Not sure where gitlint is getting run.

**Comment by GAnthony** [comment]
It seems the test is running over my commit, plus several others, as if the commit wasn't rebased.
But, I though I rebased before pushing.
Note I did do a *forced* push to update the commit (which I've done in the past with github, and seems a common method for fix branches).   If there is a different preferred method for modifying the PR, please let me know.

**Comment by galak** [comment]
Waiting for Anas to expose the shippable logs so we can get more details if we can reproduce it again.

**Pull request merged**